packages/material-ui/src/ListItem/ListItem.js

```diff
@@ -107,11 +109,23 @@ const ListItem = React.forwardRef(function ListItem(props, ref) {
     dense: dense || context.dense || false,
     alignItems,
   };
+  const listItemRef = React.useRef();
+  React.useLayoutEffect(() => {
+    if (autoFocus) {
+      listItemRef.current.focus();
```

Comment:
We should exit early if `listItemRef.current` is not set (yet) and add a warning that the rendered component can't receive focus if there's no instance. 

This is probably only relevant for function components in `component` but I don't think it's appropriate to crash just because focus isn't working properly.

Reply:
You have demonstrated how it can happen when a React.Suspense is used. Do we have to worry about the problem in this case?

Reply:
I was more concerned about users passing in function components in which case `ref.current` would always be `null`. I think we should always guard about `ref.current` being `null`. It may be obvious in the immediate context but as time goes on this could be moved farther away from the callsite and we end up questioning if this really is the case.

React never makes the claim that `ref.current` is defined at a certain point in time. Only that the instances will be set before running any effect. This includes `null`. Since there's virtually no cost for being defensive here I'd rather be safe. Especially since I don't think anybody wants their app to crash because some focus isn't properly handled.

Reply:
As long as we don't write a defensive logic to hide a core problem (that we have the leverage to fix), I'm 👍.